### PR TITLE
chore(compute): add firewall samples

### PR DESCRIPTION
**Pull request by yuryu**
Add firewall samples to google-cloud-compute-v1.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [X] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [X] Update code documentation if necessary.

closes: #16353

### CODE REVIEW

File: google-cloud-compute-v1/samples/firewall.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+# @param [String] project project ID or project number of the Cloud project you want to use.
+# @param [String] name: name of the rule that is created.
+# @param network: name of the network the rule will be applied to. Available name formats:
+#         * https://www.googleapis.com/compute/v1/projects/{project_id}/global/networks/{network}
+#         * projects/{project_id}/global/networks/{network}
+#         * global/networks/{network}
+def create_firewall_rule project:, name:, network: "global/networks/default"
+  rule = {
+    name: name,
+    direction: "INGRESS",
```
Comment (viacheslav-rostovtsev) [inline google-cloud-compute-v1/samples/firewall.rb:65]: We can use `google.cloud.compute.v1.Firewall.Direction` enum and call `to_s` on the `INGRESS` member of that to showcase that some string constants do come from an enum. Or maybe spell it out in a comment if using an enum like that is unidiomatic.

File: google-cloud-compute-v1/samples/acceptance/firewall_test.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+require_relative "helper"
+
+
+class ComputeFirewallTest < Minitest::Test
+  def setup
+    @firewalls = []
+  end
+  
+  def teardown
+    client = ::Google::Cloud::Compute::V1::Instances::Rest::Client.new
```
Comment (viacheslav-rostovtsev) [inline google-cloud-compute-v1/samples/acceptance/firewall_test.rb:27]: Should probably be `Firewalls`

**Review by viacheslav-rostovtsev — CHANGES_REQUESTED** [review]
The teardown needs to be fixed.

File: google-cloud-compute-v1/samples/firewall.rb (hunk 1 above)
Comment (yuryu) [reply google-cloud-compute-v1/samples/firewall.rb:65]: I think it's confusing to show the enum when you can't pass an enum value. I'm not sure about the long-term goal but I think we can just pass a string if it's string. The reference also says it's string. https://googleapis.dev/ruby/google-cloud-compute-v1/latest/Google/Cloud/Compute/V1/Firewall.html#direction-instance_method

File: google-cloud-compute-v1/samples/acceptance/firewall_test.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  def test_patch_priority
+    firewall_name = random_firewall_name
+    @firewalls << firewall_name
+    create_firewall_rule project: project, name: firewall_name
+    
+    client = ::Google::Cloud::Compute::V1::Firewalls::Rest::Client.new
+    firewall = client.get project: project, firewall: firewall_name
+    assert firewall.priority == 1000
```
Comment (dazuma) [inline google-cloud-compute-v1/samples/acceptance/firewall_test.rb:52]:
```ruby
assert_equal 1000, firewall.priority
```

Similar for line 55 below

**Review by dazuma — CHANGES_REQUESTED** [review]

**Review by viacheslav-rostovtsev — APPROVED** [review]

**Review by dazuma — APPROVED** [review]

**Pull request merged**